### PR TITLE
Migration to Material Components.

### PR DESCRIPTION
**Pull request by rgmf**
**Describe the pull request**
I've migrated OT to Material Components following Android's advice.

My main porpose is to use style/theme Android offer us. So, I leave colors, fonts, etc, according with Android Material Components. Anyway, I've used colors/fonts/etc fairly similar to current ones.

In that way, it's easier to maintain styles and migrate to future new frameworks.

A lot of things done:

***New Base Theme***
`Theme.MaterialComponents.DayNight.NoActionBar`

***New colors***
I've based the colors on this:
imagen

The gray colors didn't match very well with MaterialComponents unless we touch some things.

***Fonts***
I've used the styles Android Material Components offer us: https://material.io/blog/android-material-theme-type

I've followed these Nick Butcher's guidelines: https://medium.com/androiddevelopers/whats-your-text-s-appearance-f3a1729192d . Basically:
- Set any app wide styling in a textViewStyle default style in your theme.
- Set up a (small) selection of TextAppearances your app will use (or use/extend from MaterialComponent’s styles) and reference these directly from your views.
- Create styles setting any attributes not supported by TextAppearance (which themselves specify one of your TextAppearances).
- Perform any unique styling directly in your layout.

So, I've created a `type.xml` file with some typography/fonts customization based on TextAppearance following the naming recommended by Android.

***List Layout***
We were using a quite hard to understand TableLayout for marker and tracks layout that I've migrated to ConstraintLayout.

***Nigth Theme***
It's completely back with white typography. So OLED screens can save battery.

***Edit Texts***
I've used `textInputOutlinedStyle` for EditTexts.

**Screenshots**
imagen

imagen

imagen

imagen

imagen

imagen

imagen

imagen

**Link to the the issue**
#1122 
#957

**License agreement**
By opening this pull request, you are providing your contribution under the _Apache License 2.0_ (see [LICENSE.md](LICENSE.md)).

### CODE REVIEW

**Comment by rgmf** [comment]
All resources I've based on:

https://medium.com/androiddevelopers/migrating-to-material-components-for-android-ec6757795351

https://medium.com/over-engineering/setting-up-a-material-components-theme-for-android-fbf7774da739

https://medium.com/androiddevelopers/android-styling-themes-vs-styles-ebe05f917578

https://medium.com/androiddevelopers/android-styling-common-theme-attributes-8f7c50c9eaba

**Comment by dennisguse** [comment]
Screenshots look great!
The only thing that stands out is the almost invisible stop button on the first screen shot.

Colors are fine for me as well :)

File: .idea/misc.xml [hunk 1]
```diff
@@ -3,12 +3,23 @@
   <component name="DesignSurface">
     <option name="filePathToZoomLevelMap">
       <map>
-        <entry key="src/main/res/drawable/ic_activity_bike_24dp.xml" value="0.1765" />
```
Comment (dennisguse) [inline .idea/misc.xml:6]: Should be removed and we need to put this file into .gitignore.

File: src/main/java/de/dennisguse/opentracks/ui/aggregatedStatistics/FilterDialogFragment.java [hunk 2]
```diff
@@ -61,17 +62,19 @@ public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {
...
-            CheckBox checkBox = view.findViewById(R.id.filter_dialog_check_button);
-            checkBox.setText(item.value);
-            checkBox.setChecked(item.isChecked);
-            checkBox.setTag(item.id);
-            checkBox.setOnClickListener(v -> item.isChecked = !item.isChecked);
+            MaterialButton button = view.findViewById(R.id.filter_dialog_check_button);
+            button.setText(item.value);
+            button.setChecked(item.isChecked);
+            button.setTag(item.id);
+            button.setOnClickListener(v -> {
```
Comment (dennisguse) [inline src/main/java/de/dennisguse/opentracks/ui/aggregatedStatistics/FilterDialogFragment.java:75]: One line without brackets?

File: src/main/java/de/dennisguse/opentracks/ui/intervals/IntervalsFragment.java [hunk 3]
```diff
@@ -76,6 +76,15 @@ public void onCreate(@Nullable Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         stackModeListView = getArguments().getBoolean(FROM_TOP_TO_BOTTOM_KEY, true) ? IntervalStatisticsAdapter.StackMode.STACK_FROM_TOP : IntervalStatisticsAdapter.StackMode.STACK_FROM_BOTTOM;
         trackId = getArguments().getParcelable(TRACK_ID_KEY);
+        if (savedInstanceState != null) {
+            selectedInterval = (IntervalStatisticsModel.IntervalOption) savedInstanceState.getSerializable(SELECTED_INTERVAL_KEY);
```
Comment (dennisguse) [inline src/main/java/de/dennisguse/opentracks/ui/intervals/IntervalsFragment.java:80]: Feels like a functional change; can we put this into a separate commit? (I mean the bundle thing)

File: src/main/res/layout/choose_activity_type.xml [hunk 4]
```diff
@@ -31,4 +31,5 @@ limitations under the License.
         android:paddingEnd="12dp"
         android:stretchMode="spacingWidth"
         android:verticalSpacing="4dp" />
+
```
Comment (dennisguse) [inline src/main/res/layout/choose_activity_type.xml:34]: can be removed

File: src/main/res/layout/statistics_recording.xml [hunk 5]
```diff
@@ -2,7 +2,6 @@
 <androidx.constraintlayout.widget.ConstraintLayout
     xmlns:android="http://schemas.android.com/apk/res/android"
     xmlns:app="http://schemas.android.com/apk/res-auto"
-    style="@style/StatsScrollView"
```
Comment (dennisguse) [inline src/main/res/layout/statistics_recording.xml:5]: Why?

**Review by dennisguse — COMMENTED** [review]
Just skimmed through the changes: code looks good.
Will try it out tomorrow ;)

**Comment by dennisguse** [comment]
Just installed it quickly: I love the dark mode. Looks pretty awesome.
The light mode doesn't feel that well yet.

Smaller things:
* dark mode: interval list select box has the wrong background color (some other components as well?)
* light mode: TrackController background black?
* light mode: listview text color black?
* light mode: bar has the same background color as the list, but somehow that feels weird
* both: text size and text color of labels was a secondary color and less intrusive

@rgmf Anyhow: nice work :D

File: src/main/java/de/dennisguse/opentracks/ui/intervals/IntervalsFragment.java (hunk 3 above)
Comment (rgmf) [reply src/main/java/de/dennisguse/opentracks/ui/intervals/IntervalsFragment.java:80]: This change is related to the migration because the new "combo box" doesn't retain the last selected value after activity recreation (old spinner did).

File: src/main/res/layout/statistics_recording.xml (hunk 5 above)
Comment (rgmf) [reply src/main/res/layout/statistics_recording.xml:5]: We don't need it because the scroll bar will be containted into the RecyclerView but not in this ConstraintLayout.

**Comment by rgmf** [comment]
> Smaller things:
> 
>     * dark mode: interval list select box has the wrong background color (some other components as well?)
> 
>     * light mode: TrackController background black?
> 
>     * light mode: listview text color black?
> 
>     * light mode: bar has the same background color as the list, but somehow that feels weird
> 
>     * both: text size and text color of labels was a secondary color and less intrusive
> 

Good points. I'll check them out. Thank you ;)

**Comment by rgmf** [comment]
@pstorch @dennisguse It's ready.... I merge it so we can test it and check it out in the nightly build. If something is wrong or you don't like then let me know ;)
